api/database/migrations/2022_12_01_215438_add_self_declaration_fields.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+use Database\Helpers\ApiEnums;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Support\Facades\DB;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->string('indigenous_declaration_signature')->nullable()->default(null);
+            $table->jsonb('indigenous_communities')->nullable()->default(null);
+        });
+
+        DB::statement(
+            <<<SQL
+                UPDATE users
+                    SET indigenous_communities =
+                        case is_indigenous
+                            when true then :isIndigenous::jsonb
+                            when false then :emptyArray::jsonb
+                        end
+            SQL, [
+                'isIndigenous' => json_encode([ApiEnums::INDIGENOUS_LEGACY_IS_INDIGENOUS]),
+                'emptyArray' => json_encode([]), // empty array maps is_indigenous = false to the new format and enables accurate reversal
+            ]
+        );
+
+        Schema::table('users', function (Blueprint $table) {
+            $table->dropColumn('is_indigenous');
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->boolean('is_indigenous')->nullable()->default(null);
+        });
+
+        DB::statement(
+            <<<SQL
+                UPDATE users
+                    SET is_indigenous =
+                        case
+                            when indigenous_communities ?? :legacy then true
+                            when not indigenous_communities ?? :legacy then false
+                        end
+            SQL, [
+                'legacy' => ApiEnums::INDIGENOUS_LEGACY_IS_INDIGENOUS,
+            ]
+        );
```

Comment:
We should likely also set `true` if any community is selected otherwise if we rollback this migration after new users have set a community it would set `isIndigenous` to `false`

Reply:
good question hm... lets double check to make sure
@tristan-orourke 
for rolling back this migration, should the rollback set `is_indigenous = true` if the array of communities is non-zero length or only if the array contains the `LEGACY` enum?

Reply:
Tristan suggests "non-zero"

Reply:
done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/4994/commits/4aee11c04cb29c2e779bcca9a7bb89dcd1b41ca0